"Unexpected indentation" warnings from :members:

Since Sphinx updated to 1.7.0, our SymPy docs started failing. We have the setting enabled in our docs to turn warnings into errors. 

We have gotten many errors like

```
Warning, treated as error:
/home/travis/build/sympy/sympy/sympy/codegen/cfunctions.py:docstring of sympy.codegen.cfunctions.exp2.eval:15:Unexpected indentation.
make: *** [html] Error 2
```

It turns out the source of the error has nothing to do with the method in question (which doesn't even have a docstring). The problem is that we used

```
.. automodule:: sympy.codegen.cfunctions
    :members:
```

instead of

```
.. automodule:: sympy.codegen.cfunctions

   :members:
```

This is requiring changes across our entire docs.

Now if Sphinx wants to require the newline here, that is fine. But the warning should be much more indicative of what the problem is. Also, quite annoyingly, it only warns about one issue at a time, so if you fix an issue and rebuild, it will warn about another one (and for whatever reason, Sphinx no longer caches the build??? Every build is done from scratch. This is even if I make no changes to anything). 

I tested this against the master branch and it is the same result. 

Upstream SymPy issue: https://github.com/sympy/sympy/pull/14181

--- a/CHANGES
+++ b/CHANGES
@@ -20,6 +20,8 @@ Features added
 Bugs fixed
 ----------
 
+* #4606: autodoc: the location of the warning is incorrect for inherited method
+
 Testing
 --------
 

--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -543,9 +543,18 @@ class Documenter:
             yield from docstringlines
 
     def get_sourcename(self) -> str:
+        if (getattr(self.object, '__module__', None) and
+                getattr(self.object, '__qualname__', None)):
+            # Get the correct location of docstring from self.object
+            # to support inherited methods
+            fullname = '%s.%s' % (self.object.__module__, self.object.__qualname__)
+        else:
+            fullname = self.fullname
+
         if self.analyzer:
-            return '%s:docstring of %s' % (self.analyzer.srcname, self.fullname)
-        return 'docstring of %s' % self.fullname
+            return '%s:docstring of %s' % (self.analyzer.srcname, fullname)
+        else:
+            return 'docstring of %s' % fullname
 
     def add_content(self, more_content: Any, no_docstring: bool = False) -> None:
         """Add content from docstrings, attribute documentation and user."""

